### PR TITLE
Add style guide with destructuring of props and components with context (atomic design)

### PR DESCRIPTION
**Pull request by aaranadev**

### CODE REVIEW

**Comment by alasarr** [comment]
Please, add a title at the PR, it's a public repo and the link to CH is not enough.

**Comment by alasarr** [comment]
A description is also welcome

File: template-skeleton/template/src/App.js [hunk 1]
```diff
@@ -36,24 +36,24 @@ theme = responsiveFontSizes(theme, {
 });
 
 const useStyles = makeStyles(() => ({
-  root: {
-    flex: 1,
+  app: {
```
Comment (alasarr) [inline template-skeleton/template/src/App.js:39]: More than the target of this PR

File: template-skeleton/template/src/components/common/ErrorSnackbar.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Snackbar } from '@material-ui/core';
```
Comment (alasarr) [inline template-skeleton/template/src/components/common/ErrorSnackbar.js:1]: It exceeds the feature scope of the PR

File: template-skeleton/template/src/components/common/Header.js [hunk 3]
```diff
@@ -17,75 +17,101 @@ import MenuIcon from '@material-ui/icons/Menu';
...
-    '& .MuiTabs-indicator': {
-      backgroundColor: theme.palette.common.white,
-    },
-  },
-  divider: {
-    margin: theme.spacing(0, 3),
-  },
-  menuButton: {
-    margin: theme.spacing(0, 0.75, 0, -1.25),
+import { ReactComponent as CartoLogo } from 'assets/img/carto-logo.svg';
```
Comment (alasarr) [inline template-skeleton/template/src/components/common/Header.js:20]: Ditto

File: template-skeleton/template/src/components/common/Header.js [hunk 4]
```diff
@@ -17,75 +17,101 @@ import MenuIcon from '@material-ui/icons/Menu';
...
-  drawer: {
-    minWidth: 260,
-  },
+const useStylesCommon = makeStyles((theme) => ({
   title: {
     '& h1': {
+      display: 'flex',
       fontWeight: theme.typography.fontWeightRegular,
-      color: theme.palette.common.white,
+      color: theme.palette.appBar?.contrastText || theme.palette.primary?.contrastText,
```
Comment (alasarr) [inline template-skeleton/template/src/components/common/Header.js:28]: ditto

File: template-skeleton/template/src/components/common/Header.js [hunk 5]
```diff
@@ -17,75 +17,101 @@ import MenuIcon from '@material-ui/icons/Menu';
...
-        <Tabs
-          value={location.pathname.split('/')[1] || ''}
-          textColor={vertical ? 'primary' : 'inherit'}
-          orientation={vertical ? 'vertical' : 'horizontal'}
-          variant={vertical ? 'fullWidth' : 'standard'}
-        >
-          <Tab label='Home' value='' component={NavLink} to='/' />
-          {/* [hygen] Import links */}
-        </Tabs>
+    <Hidden xsDown>
```
Comment (alasarr) [inline template-skeleton/template/src/components/common/Header.js:81]: ditto

File: template-skeleton/template/src/components/layers/index.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 // [hygen] Import layers
 
-export const getLayers = () => {
+export default function getLayers() {
```
Comment (alasarr) [inline template-skeleton/template/src/components/layers/index.js:3]: ditto

**Review by alasarr — CHANGES_REQUESTED** [review]
This PR exceeds the feature specified in the description: styles, new components, etc.

Add only in this PR the feature of destructuring props.

**Comment by aaranadev** [comment]
@alasarr I leaved a note in the task: https://app.clubhouse.io/cartoteam/story/160574/improve-the-css-by-default-including-names-and-theme-file.

I've modified the files and the styles, I can't change the style only in other PR without conficts

**Comment by alasarr** [comment]
>I've modified the files and the styles, I can't change the style only in other PR without conflicts

Create the style PR from this one. And add a note that this PR needs to be merged before

**Comment by aaranadev** [comment]
Perfect

**Comment by alasarr** [comment]
Steps:

1. Create a PR with only the features of this.
2. Create another PR with the improvement of styles and explain in the title and description what you're doing

**Comment by aaranadev** [comment]
I've created several PR from this with minor improvements.

I close this PR

**Pull request closed without merging**